### ISSUE TITLE
GR.OPTION_FLIP_Y causes weird pixel spacing in drawimage

### ISSUE THREAD

**Issue by jusack** [jheinen/GR.jl#527]
I encountered a weird bug in https://github.com/JuliaPlots/Plots.jl/issues/4833 which I traced to the yflip parameter. I was able to reproduce it in a MWE using only GR. Do you have any idea, what might be the cause for the observed behavior? I am at version GR v0.72.10 and GR_jll v0.72.10+0

```julia
using GR
GR.clearws()
GR.setviewport(0,1,0,1)
GR.setwindow(0,50,0,50)
GR.setscale(GR.OPTION_FLIP_Y) # GR.setscale(0) is fine
GR.drawimage(0,300,300,0,300,300,rand(UInt32,90000))
GR.updatews()
```

**Comment by jheinen** [comment]
Could you please explain what the problem is? I tried the following example (with a real image) and could not see any problems:

```julia
using GR

w, h, d = readimage("tiger.png")

setviewport(0, 1, 0, 1)
setwindow(0, 50, 0, 50)

setscale(GR.OPTION_FLIP_Y)

drawimage(0, 100, 0, 100, w, h, d)
updatews()
```

Input image (tiger.png):
![tiger](https://github.com/jheinen/GR.jl/assets/5718244/b4da69bf-9a2c-4706-875e-9cee75ddadec)


Result:
<img width="612" alt="Screenshot 2023-10-11 at 11 01 08" src="https://github.com/jheinen/GR.jl/assets/5718244/ffec24eb-4596-4e4e-b38f-e30538c5505c">

**Comment by jusack** [comment]
I have posted some images of how it looks in the original issue in Plots linked above. This is the result of the pure GR version: 
![grafik](https://github.com/jheinen/GR.jl/assets/38977690/306c7b44-5b8f-4482-9174-4cb425393efb).
In Plots.jl I observed that it depends on the size of the image that is plotted, if I crop the tiger.png to 300x300 (a size that I observed to show the pixel unevenness strongly)
![tiger](https://github.com/jheinen/GR.jl/assets/38977690/4484f02a-be64-462b-b58d-298fdec30994)
 I can also see the unevenness there: (using your code snippet, shown side by side with MS Paint) 
![grafik](https://github.com/jheinen/GR.jl/assets/38977690/75a9df67-b15b-4c1c-ac79-71985eeb9929)

**Comment by IngoMeyer441** [comment]
@jusack Thanks for your pointing us to this bug. We fixed the non-square pixels in [C-GR commit b9aefa6](https://github.com/sciapp/gr/commit/b9eafa6e8e46bde3936995e89cd031a7413daa4c), so (currently) you will need to use the latest GR runtime to get the fix.
